### DISCUSSION TITLE
two-way binding, v-model, and form handling

### DISCUSSION THREAD

**Discussion by plweil** [vuejs/pinia#854] [Help and Questions] [answered]
Regarding the Vuex documentation on [Form Handling](https://vuex.vuejs.org/guide/forms.html) , does Pinia change this in any way? Right now under Vuex, I rely pretty heavily on [vuex-map-fields](https://github.com/maoberlehner/vuex-map-fields/) to handle this (form fields in the store) and avoid extensive two-way computed properties. If I want to migrate to Pinia, what is the recommended way to handle this?

**Comment by posva** [comment, marked as answer]
In pinia you don't need any extra lib:

```js
const useStore = defineStore(id, { state: () => ({ name: '' }) })

const store = useStore()
```

```vue
<input v-model="store.name">
```

**Reply by plweil** [reply]
Great!! Thanks @posva .

**Reply by lucpotage** [reply]
Thanks @posva.

**Reply by aacassandra** [reply]
Awesome!!! Thanks @posva

**Reply by lucas-aguilar** [reply]
Thanks a lot!

**Reply by YutoYasunaga** [reply]
That's great, very easy to use. Thanks!

**Reply by ghost** [reply]
Woah, amazing!

**Comment by trev-dev** [comment]
Is this the same case using the "options" api, where "mapState" is being used?

Edit: use `mapWritableState` ✅
